frontend/src/queries/query.ts

```diff
@@ -113,44 +114,58 @@ export async function query<N extends DataNode = DataNode>(
+        } else if (isInsightQueryNode(queryNode)) {
+            const filters = queryNodeToFilter(queryNode)
+            const params = {
+                ...filters,
+                ...(refresh ? { refresh: true } : {}),
+                client_query_id: queryId,
+                session_id: currentSessionId(),
+            }
+            const [resp] = await legacyInsightQuery({
+                filters: params,
+                currentTeamId: getCurrentTeamId(),
+                methodOptions,
+                refresh,
+            })
+            response = await resp.json()
+        } else if (isLegacyQuery(queryNode)) {
+            const [resp] = await legacyInsightQuery({
+                filters: queryNode.filters,
+                currentTeamId: getCurrentTeamId(),
+                methodOptions,
+            })
+            response = await resp.json()
+        } else if (isTimeToSeeDataQuery(queryNode)) {
+            response = await api.query(
+                {
+                    ...queryNode,
+                    teamId: queryNode.teamId ?? getCurrentTeamId(),
+                    sessionId: queryNode.sessionId ?? currentSessionId(),
+                    sessionStart: queryNode.sessionStart ?? now().subtract(1, 'day').toISOString(),
+                    sessionEnd: queryNode.sessionEnd ?? now().toISOString(),
+                },
+                methodOptions
+            )
+        } else {
+            response = await api.query(queryNode, methodOptions, queryId)
+            if (isHogQLQuery(queryNode) && response && typeof response === 'object') {
+                logParams.clickhouse_sql = (response as HogQLQueryResponse)?.clickhouse
+            }
         }
-        const [response] = await legacyInsightQuery({
-            filters: params,
-            currentTeamId: getCurrentTeamId(),
-            methodOptions,
-            refresh,
-        })
-        return await response.json()
-    } else if (isLegacyQuery(queryNode)) {
-        const [response] = await legacyInsightQuery({
-            filters: queryNode.filters,
-            currentTeamId: getCurrentTeamId(),
-            methodOptions,
-        })
-        return await response.json()
-    } else if (isTimeToSeeDataQuery(queryNode)) {
-        return await api.query(
-            {
-                ...queryNode,
-                teamId: queryNode.teamId ?? getCurrentTeamId(),
-                sessionId: queryNode.sessionId ?? currentSessionId(),
-                sessionStart: queryNode.sessionStart ?? now().subtract(1, 'day').toISOString(),
-                sessionEnd: queryNode.sessionEnd ?? now().toISOString(),
-            },
-            methodOptions
-        )
+        posthog.capture('Query', { query: queryNode, duration: performance.now() - startTime, ...logParams })
+        return response
+    } catch (e) {
+        posthog.capture('Query Error', { query: queryNode, duration: performance.now() - startTime, ...logParams })
```

Comment:
Nit: I think it's quite rare for our events to look like "Title Case", they're usually more something along the lines of `query completed` or `query failed`

Reply:
makes sense, renamed